sonic_platform_base/module_base.py

```diff
@@ -196,6 +214,67 @@ def get_maximum_consumed_power(self):
         """
         raise NotImplementedError
 
+    ##############################################
+    # SmartSwitch methods
+    ##############################################
+
+    def get_dpu_id(self):
+        """
+        Retrieves the DPU ID. Returns None for non-smartswitch chassis.
+
+        Returns:
+            An integer, indicating the DPU ID. DPU0 returns 0, DPUX returns X
+        """
+        raise NotImplementedError
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the cause of the previous reboot of the DPU module
+
+        Returns:
+            A tuple (string, string) where the first element is a string
+            containing the cause of the previous reboot. This string must
+            be one of the predefined strings in this class. If the first
+            string is "REBOOT_CAUSE_HARDWARE_OTHER", the second string can be
+            used to pass a description of the reboot cause.
+
+        """
+        raise NotImplementedError
+
+    def get_state_info(self):
+        """
+        Retrieves the dpu state object having the detailed dpu state progression.
+        Fetched from ChassisStateDB.
+
+        Returns:
+            An object instance of the DPU_STATE (see DB schema)
+            Returns None on switch module
+
+        Sample Output: {
```

Comment:
A small suggestion here is not changes the object to the following:
```
{
    'dpu_control_plane': {'state': 'UP', 'time': '20240626 21:13:25', 'reason': '...'},
    'dpu_data_plane': {'state': 'UP', 'time': '20240626 21:13:25', 'reason': '...'},
    'dpu_midplane_link': {'state': 'UP', 'time': '20240626 21:13:25', 'reason': '...'}
}
``` 
This simplifies the implementation and makes the object compact.